examples/dataflow-flex-python/df-package/corder/customer_orders.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import json
+import logging
+
+import apache_beam as beam
+from apache_beam.io import fileio
+
+from corder.bq_schema import CUSTOMER_TABLE_SCHEMA, ORDER_TABLE_SCHEMA
+
+
+def parse_to_dict(file):
+    from xsdata.formats.dataclass.parsers import XmlParser
+    from corder.customer_orders_model import Root
+    parser = XmlParser()
+    root = parser.from_bytes(file.read(), Root)
+    yield root
+
+
+def get_orders(root):
+    from xsdata.formats.dataclass.serializers import JsonSerializer
+    for order in root.orders.order:
```

Comment:
Note that while these elements are required, the model isn't validated on deserialization.
You can still end up with `root.orders` as `None` if a malformed document makes its way through the pipeline, because the model states the default value for `orders` is `None`.
To counter that I think you can use xsdata with attrs or pydantic (xsdata-attrs, xsdata-pydantic) to make sure that the model is validated or else an exception is thrown.
You'd have to handle the error by sending the element to a dead letter side output or by dropping it.

Reply:
I added pydantic models and tested that it validates field type. But still all field are getting set as Optional with default values. I found [this](https://github.com/tefra/xsdata/issues/555#issuecomment-879654328) explanation about the behaviour. xsdata-attrs can validate it but only from python 3.10 onwards. We could have written some custom code to verify if some field is required and its value is None then write that to dead letter queue, but I think its out of the scope of the commit.  I added None type check in pipeline though.
Please let me know your thoughts.